src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1171,9 +1171,19 @@ MessageHelper::CreateGlobalPropertiesRequestsToHMI(
     return requests;
   }
 
+  auto& help_prompt_manager =
+      const_cast<Application*>(app.get())->help_prompt_manager();
+
+  const bool can_send_ui = helpers::Compare<HelpPromptManager::SendingType,
+                                            helpers::EQ,
+                                            helpers::ONE>(
+      help_prompt_manager.GetSendingType(),
+      HelpPromptManager::SendingType::kSendVRHelp,
+      HelpPromptManager::SendingType::kSendBoth);
+
   // UI global properties
 
-  if (app->vr_help_title() || app->vr_help()) {
+  if (can_send_ui && (app->vr_help_title() || app->vr_help())) {
```

Comment:
Why would you restrict an app from setting this value twice?

Reply:
@jacobkeeler it is required for case when SDL is resuming some application commands and at this point of time SDL receives `SetGlobalProperties` from the same application. In this case SDL should not send resumed commands via `SetGlobalProperties` as application sets the custom one